src/DefifaGovernor.sol

```diff
@@ -158,6 +181,28 @@ contract DefifaGovernor is
         return super.proposalThreshold();
     }
 
+    function verifyProofsAndExecute(
+        uint256 proposalId,
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash,
+        bytes32[] calldata _leaves
+    ) external {
+        bytes32 _root = proposalScorecardRoot[proposalId];
+        socrecardVerifier.verifyScorecard(_leaves, _root);
```

Comment:
Why are we verifying the scorecard if we just set/generated it and we are not using the leaves

Reply:
so the flow is when a proposal is created we generate a root against that proposal id and we emit an [event](https://github.com/jbx-protocol/juice-defifa/blob/feat/scorecard-verifier/src/DefifaScoreCardVerifier.sol#L41) so we have the leaves but while execution someone can generate leaves from an invalid scorecard and call `verifyProofsAndExecute`  so to prevent this we need to generate the proof and verify it against the root
does this make sense to you ?